src/Core/Content/Product/Cart/ProductFeatureBuilder.php

```diff
@@ -240,9 +256,10 @@ private function getCustomField(string $name, CartDataCollection $data, SalesCha
         }
 
         $customField = $data->get($fieldKey);
-        $label = $this->getCustomFieldLabel($customField);
+        \assert($customField instanceof CustomFieldEntity);
```

Comment:
maybe do a real if condition instead :thinking: or can we really be sure, this is always true?

Reply:
Mh, so the data key is `custom-field-{$customFieldName}` and it has been checked that is exists. So it actually should always be a `CustomFieldEntity` otherwise something is broken.

Reply:
if condition that then throws an exception maybe?

Reply:
Mh, okay :-D I changed the behavior to throw an exception.